### PR TITLE
cleanup floating point exception management

### PR DESCRIPTION
**Pull request by parejkoj**
Cleanup floating point exception management, but keep it as GNU only. May be useful for future debugging, but the "portable" version of fenv.h that I proposed in RFC-158 was not as portable as I'd hoped.

### CODE REVIEW

File: tests/test_wcs.cc [hunk 1]
```diff
@@ -12,26 +12,26 @@
 #include "lsst/afw/image/TanWcs.h"
 #include "lsst/afw/image/Utils.h"
 #include "lsst/afw/fits.h"
-#include "lsst/daf/base.h" 
+#include "lsst/daf/base.h"
 
 #include <stdlib.h> /* for getenv */
 
+// NOTE: turn this flag on to raise exceptions on floating point errors.
+// NOTE: this only works on GNU/Linux (fenv is not C++ standard).
```
Comment (timj) [inline tests/test_wcs.cc]: `fenv.h` is a standard include file from the C99 floating point environment. What's non-standard is the Gnu `feenableexcept` function.

File: tests/test_wcs.cc [hunk 2]
```diff
@@ -12,26 +12,26 @@
...
-  /* Enable some exceptions.  At startup all exceptions are masked.  */
-  
-  if (getenv("DUMP_CORE_ON_FPE"))
-    feenableexcept (FE_INVALID|FE_DIVBYZERO|FE_OVERFLOW);
+   // Enable some exceptions.  At startup all exceptions are masked.
+  feenableexcept (FE_INVALID|FE_DIVBYZERO|FE_OVERFLOW);
 }
- 
+#endif
+
```
Comment (timj) [inline tests/test_wcs.cc]: Don't need this extra blank line

File: tests/test_wcs.cc [hunk 3]
```diff
@@ -12,26 +12,26 @@
 #include "lsst/afw/image/TanWcs.h"
 #include "lsst/afw/image/Utils.h"
 #include "lsst/afw/fits.h"
-#include "lsst/daf/base.h"
```
Comment (timj) [inline tests/test_wcs.cc]: Can you do whitespace fixes in a separate commit?

File: tests/test_wcs.cc (hunk 1 above)
Comment (parejkoj) [inline tests/test_wcs.cc]: Fixed and pushed.

File: tests/test_wcs.cc (hunk 3 above)
Comment (parejkoj) [inline tests/test_wcs.cc]: I'll do a bulk whitespace pass on it now, to avoid this problem in the future.

**Comment by timj** [comment]
I think that commit history should have been sorted out before merging. In particular that fake merge commit. :wink:

**Comment by parejkoj** [comment]
That was me attempting to fix it. Apparently I failed.
